### PR TITLE
Some other NoSuchMethod, NPEs fixed

### PR DESCRIPTION
**Pull request by Rafael2616**
## Description

this fixes this issue also, when clciking in the widtget without holding it

before:
https://github.com/user-attachments/assets/a4f1bb16-5841-42e8-9dd8-07f7125c064a

after:
nothing will happens if only click in the widget

Fixes #(issue) 

## Type of change

:x: General change (non-breaking change that doesn't fit the below categories like copyediting)
:white_check_mark: Bug fix (non-breaking change which fixes an issue)
:x: New feature (non-breaking change which adds functionality)
:x: Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

File: src/com/android/launcher3/widget/BaseWidgetSheet.java [hunk 1]
```diff
@@ -163,12 +163,12 @@ public final void onClick(View v) {
             if (mWidgetCellWithAddButton != wc) {
                 // If click is on a cell not showing an add button, show it now.
                 final PendingAddItemInfo info = (PendingAddItemInfo) wc.getTag();
-                if (mActivityContext instanceof Launcher) {
-                    wc.showAddButton((view) -> addWidget(info));
-                } else {
-                    wc.showAddButton((view) -> mActivityContext.getItemOnClickListener()
-                            .onClick(wc));
-                }
+//                if (mActivityContext instanceof Launcher) {
```
Comment (MrSluffy) [inline src/com/android/launcher3/widget/BaseWidgetSheet.java]: why are we commenting this? is there no work arround?

**Comment by MrSluffy** [comment]
and the widget shortcut icons seems to be broken, I might need to check this

**Comment by Rafael2616** [comment]
> and the widget shortcut icons seems to be broken, I might need to check this

This is not related to this PR, this issue was already present

File: src/com/android/launcher3/widget/BaseWidgetSheet.java (hunk 1 above)
Comment (Rafael2616) [reply src/com/android/launcher3/widget/BaseWidgetSheet.java]: this is the code that makes that artifact appear when clicking on widgets without pressing them, I actually don't know why this is useful, so I just commented it so I don't lose sight of it

The behavior is shown in the video

Comment (validcube) [reply src/com/android/launcher3/widget/BaseWidgetSheet.java]: > this is the code that makes that artifact appear when clicking on widgets without pressing them, I actually don't know why this is useful, so I just commented it so I don't lose sight of it
> 
> The behavior is shown in the video

It's likely part of Android 15 feature here

image

Comment (Rafael2616) [reply src/com/android/launcher3/widget/BaseWidgetSheet.java]: Thanks for the info, I'll update the PR taking this into consideration

Comment (Rafael2616) [reply src/com/android/launcher3/widget/BaseWidgetSheet.java]: Do you know if this is only available in Android 15, or is it also present in Android 14 or other versions?

File: src/com/android/launcher3/pm/InstallSessionHelper.java [hunk 2]
```diff
@@ -262,6 +262,6 @@ public InstallSessionTracker registerInstallTracker(
     }
 
     public static UserHandle getUserHandle(@NonNull final SessionInfo info) {
-        return info.getUser();
+        return Utilities.ATLEAST_Q ? info.getUser() : UserHandle.getUserHandleForUid(info.getSessionId());
```
Comment (MrSluffy) [inline src/com/android/launcher3/pm/InstallSessionHelper.java]: Please use Process.myUserHandle() or UserHandle.getUserHandleForUid(info.userId) instead

Comment (Rafael2616) [reply src/com/android/launcher3/pm/InstallSessionHelper.java]: Right

File: src/com/android/launcher3/widget/BaseWidgetSheet.java (hunk 1 above)
Comment (Rafael2616) [reply src/com/android/launcher3/widget/BaseWidgetSheet.java]: @MrSluffy should i isolate this by Android Version ?? as i thinkt this is not expected to work before a15?

**Comment by Rafael2616** [comment]
i just revert the changes in `BaseWidgetSheet` 
so that only changes related to crashes remain in this PR @MrSluffy can you check this

File: src/com/android/launcher3/widget/BaseWidgetSheet.java (hunk 1 above)
Comment (validcube) [reply src/com/android/launcher3/widget/BaseWidgetSheet.java]: > Do you know if this is only available in Android 15, or is it also present in Android 14 or other versions?

Hello, sorry for the slow response. This feature or "hint" is only available in Android 15 (as of Beta 2)

File: src/com/android/launcher3/compat/AccessibilityManagerCompat.java [hunk 3]
```diff
@@ -110,6 +112,9 @@ private static AccessibilityManager getAccessibilityManagerForTest(Context conte
     }
 
     public static int getRecommendedTimeoutMillis(Context context, int originalTimeout, int flags) {
-        return getManager(context).getRecommendedTimeoutMillis(originalTimeout, flags);
+        if (ATLEAST_Q) {
+            return getManager(context).getRecommendedTimeoutMillis(originalTimeout, flags);
```
Comment (MrSluffy) [inline src/com/android/launcher3/compat/AccessibilityManagerCompat.java]: use ternary instead return AT LEAST Q ? getManager(context).getRecommendedTimeoutMillis(originalTimeout, flags) : originalTimeout

**Review by MrSluffy — CHANGES_REQUESTED** [review]

File: src/com/android/launcher3/compat/AccessibilityManagerCompat.java (hunk 3 above)
Comment (Rafael2616) [reply src/com/android/launcher3/compat/AccessibilityManagerCompat.java]: done

**Review by MrSluffy — APPROVED** [review]

**Pull request merged**